deno/payloads/v10/_interactions/base.ts

```diff
@@ -129,3 +129,36 @@ export type APIGuildInteractionWrapper<Original extends APIBaseInteraction<Inter
 	'user'
 > &
 	Required<Pick<Original, 'member' | 'guild_id'>>;
+
+/**
+ * https://discord.com/developers/docs/resources/channel#channel-object
+ */
+export interface APIInteractionDataResolvedChannel extends Required<APIPartialChannel> {
+	thread_metadata?: APIThreadMetadata | null;
+	permissions: Permissions;
+	parent_id?: string | null;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/guild#guild-member-object
+ */
+export interface APIInteractionDataResolvedGuildMember extends Omit<APIGuildMember, 'user' | 'deaf' | 'mute'> {
+	permissions: Permissions;
+}
+
+/**
+ * https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-resolved-data-structure
+ */
+export interface APIInteractionDataResolved {
+	users?: Record<Snowflake, APIUser>;
+	roles?: Record<Snowflake, APIRole>;
+	members?: Record<Snowflake, APIInteractionDataResolvedGuildMember>;
+	channels?: Record<Snowflake, APIInteractionDataResolvedChannel>;
+	attachments?: Record<Snowflake, APIAttachment>;
+}
+
+/**
+ * `users` and optional `members` from APIInteractionDataResolved, for user commands and user selects
+ */
+export type APIUserInteractionDataResolved = Required<Pick<APIInteractionDataResolved, 'users'>> &
```

Comment:
The name changes make this PR a major change, can you please add deprecated aliases to the old names? 🙏